multimodal-search-pdf/pods/segmenters.py

```diff
@@ -0,0 +1,285 @@
+import io
+from typing import Dict, List
+
+import numpy as np
+from jina.executors.segmenters import BaseSegmenter
+from jina.executors.decorators import single, batching
+
+
+class PDFSegmenter(BaseSegmenter):
+    """
+    :class:`PDFExtractorSegmenter` Extracts data (text and images) from PDF files.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=3)
```

Comment:
As it is `single`, there is no need to use `slice_nargs`
```suggestion
    @single
```

Reply:
remove this will not work I think. I tried to remove it but then no results returned.

Reply:
And this parameter should be set  to equal to the number of arguments if number of the arguments is larger than 1

Reply:
hmm, that is interesting. I will dig into this

Reply:
@Yongxuanzhang is right. Single speaks about instances passed, not about the amounts of arguments